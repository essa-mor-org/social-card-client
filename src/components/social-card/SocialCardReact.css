.Social_react{
	display: flex;
	justify-content: space-between;
	padding-top: var(--Social-space);
	padding-bottom: var(--Social-space);
	padding-left: calc(var(--Social-space) * 6);
	padding-right: calc(var(--Social-space) * 6);
	border-bottom: 1px solid var(--Social-border-color);
}