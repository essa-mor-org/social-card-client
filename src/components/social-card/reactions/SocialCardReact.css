.Social_react{
	display: flex;
	justify-content: space-between;
	padding-top: var(--Social-space);
	padding-bottom: var(--Social-space);
	padding-left: calc(var(--Social-space) * 6);
	padding-right: calc(var(--Social-space) * 6);
	border-bottom: 1px solid var(--Social-border-color);
}

.Reaction_like_text_active{
	color: rgb(53, 120, 229);
}