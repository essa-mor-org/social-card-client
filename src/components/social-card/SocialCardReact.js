import React from 'react';
import PropTypes from 'prop-types';

import ReactionButton from '../buttons/ReactionButton';

import './SocialCardReact.css';

function SocialCardReact({onLike, onComment, onShare}){
	return (<div className="Social_react">
		<ReactionButton text="Like" className="Reaction_like" onClick={onLike} />
		<ReactionButton text="Comment" className="Reaction_comment" onClick={onComment}/>
		<ReactionButton text="Share" className="Reaction_share" onClick={onShare}/>
	</div>);
}

SocialCardReact.propTypes = {
	onLike: PropTypes.func.isRequired,
	onComment: PropTypes.func.isRequired,
	onShare: PropTypes.func.isRequired
};

export default SocialCardReact;