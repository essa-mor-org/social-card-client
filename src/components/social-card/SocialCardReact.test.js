import React from 'react';
import SocialCardReact from './SocialCardReact';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';

describe('SocialCardReact test', () => {
	it('renders correctly', () => {
		const tree = renderer
			.create(<SocialCardReact onLike={jest.fn()} onComment={jest.fn()} onShare={jest.fn()}/>)
			.toJSON();
		expect(tree).toMatchSnapshot();
	});

	it('renders correctly - click callback', () => {
		const likeCallBack = jest.fn();
		const commentCallBack = jest.fn();
		const shareCallBack = jest.fn();
		const wrapper = shallow(<SocialCardReact 
			onLike={likeCallBack} 
			onComment={commentCallBack}
			onShare={shareCallBack}/>);
		wrapper.find('.Reaction_like').simulate('click');
		expect(likeCallBack).toHaveBeenCalledTimes(1);
		wrapper.find('.Reaction_comment').simulate('click');
		expect(commentCallBack).toHaveBeenCalledTimes(1);
		wrapper.find('.Reaction_share').simulate('click');
		expect(shareCallBack).toHaveBeenCalledTimes(1);
	});
});
