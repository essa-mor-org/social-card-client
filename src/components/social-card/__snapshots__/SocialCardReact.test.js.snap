// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SocialCardReact test renders correctly 1`] = `
<div
  className="Social_react"
>
  <a
    className="Reaction_like Reaction_button"
    href="#"
    onClick={[Function]}
  >
    Like
  </a>
  <a
    className="Reaction_comment Reaction_button"
    href="#"
    onClick={[Function]}
  >
    Comment
  </a>
  <a
    className="Reaction_share Reaction_button"
    href="#"
    onClick={[Function]}
  >
    Share
  </a>
</div>
`;
